File: gaia-sdk-python/gaia_sdk/graphql/request/type/TopExecutedBehaviour.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class TopExecutedBehaviour(list):

    def behaviour_id(self):
        self.append(lambda x: "behaviourId")

    def behaviour_name(self):
        self.append(lambda x: "behaviourName")

    def number_of_executions(self):
        self.append(lambda x: "numberOfExecutions")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/TopExecutedBehaviour.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class TopExecutedBehaviour extends Array<(_:VariableRegistry) => string> {
public _typeName = "TopExecutedBehaviour";
    public behaviourId = () => { 
        this.push(_ => "behaviourId")
    };

    public behaviourName = () => { 
        this.push(_ => "behaviourName")
    };

    public numberOfExecutions = () => { 
        this.push(_ => "numberOfExecutions")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

